src/API/HealthCheck.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Return Pinterest Feed health status.
+ *
+ * @package     Pinterest_For_Woocommerce/API
+ * @version     1.0.0
+ */
+
+namespace Automattic\WooCommerce\Pinterest\API;
+
+use Automattic\WooCommerce\Pinterest as Pinterest;
+
+use \WP_REST_Server;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Endpoint used to check the Health status of the connected Merchant object.
+ */
+class HealthCheck extends VendorAPI {
+
+	/**
+	 * Initialize class
+	 */
+	public function __construct() {
+		$this->base              = 'health';
+		$this->endpoint_callback = 'health_check';
+		$this->methods           = WP_REST_Server::READABLE;
+
+		$this->register_routes();
+	}
+
+
+	/**
+	 * Get the merchant object from the API and return the status, and if exists, the dissapproval rationale.
+	 *
+	 * @return array
+	 *
+	 * @throws \Exception PHP Exception.
+	 */
+	public function health_check() {
+
+		try {
+
+			if ( ! Pinterest_For_Woocommerce()::get_data( 'merchant_id' ) ) {
+				return array(
+					'status' => 'pending_initial_configuration',
+				);
+			}
+
+			$merchant = Pinterest\Merchants::get_merchant();
+
+			if ( 'success' !== $merchant['status'] || empty( $merchant['data']->product_pin_approval_status ) ) {
+				throw new \Exception( esc_html__( 'Could not get approval status from Pinterest.', 'pinterest-for-woocommerce' ), 200 );
```

Comment:
I see that this is being done elsewhere, but given that this is an API response, I don't believe `esc_html__()` is the correct function to use here. Instead, the normal `__()` function should be used instead.

```suggestion
				throw new \Exception( __( 'Could not get approval status from Pinterest.', 'pinterest-for-woocommerce' ), 200 );
```